chore(core): name all scenario pkgs

boneskull opened the pull request:
`@endo/compartment-mapper` cannot ingest scenarios if the `package.json` files therein do not have `name` fields.  This change adds the fields.

boneskull committed:
chore(core): name all scenario pkgs
`@endo/compartment-mapper` cannot ingest scenarios if the `package.json` files therein do not have `name` fields.  This change adds the fields.

kumavis reviewed (approved):
noticed CI test is failing

boneskull committed:
chore(core): add fs parameter to prepareScenarioOnDisk

This allows us to swap a replacement `fs` in tests to run against a virtual filesystem.

boneskull committed:
fix(core): add better types to policy loading and merging functions

This also refactors these functions to use `node:fs/promises` instead of synchronous operations. `fs.existsSync` is avoided due to the extra filesystem operation on the happy path.
